game/src/contrib/hud/crafting/CraftingGUI.java

```diff
@@ -0,0 +1,342 @@
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.Batch;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.graphics.g2d.GlyphLayout;
+import com.badlogic.gdx.graphics.g2d.TextureRegion;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.scenes.scene2d.utils.DragAndDrop;
+import com.badlogic.gdx.utils.Align;
+
+import contrib.components.InventoryComponent;
+import contrib.crafting.Crafting;
+import contrib.crafting.CraftingResult;
+import contrib.crafting.CraftingType;
+import contrib.crafting.Recipe;
+import contrib.hud.CombinableGUI;
+import contrib.hud.GUICombination;
+import contrib.hud.ImageButton;
+import contrib.hud.inventory.ItemDragPayload;
+import contrib.utils.components.item.ItemData;
+
+import core.Game;
+import core.utils.components.draw.Animation;
+import core.utils.components.draw.TextureMap;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/** */
+public class CraftingGUI extends CombinableGUI {
+
+    // Position settings
+    private static final int NUMBER_PADDING = 5;
+    private static final int ITEM_GAP = 10;
+
+    // Positioning and sizing
+    // These values should fit the background texture of the crafting GUI and should be between 0
+    // and 1.
+    // 0 is the bottom left corner and 1 is the top right corner.
+
+    // X coordinate of the center of the input item row.
+    private static final float INPUT_ITEMS_X = 0.5f;
+
+    // Y coordinate of the bottom edge of the input item row.
+    private static final float INPUT_ITEMS_Y = 0.775f;
+
+    // The size is based on the height of the crafting GUI and items are always square.
+    private static final float INPUT_ITEMS_MAX_SIZE = 0.2f;
+
+    // X coordinate of the center of the result item.
+    private static final float RESULT_ITEM_X = 0.5f;
+
+    // Y coordinate of the bottom edge of the result item.
+    private static final float RESULT_ITEM_Y = 0.219f;
+
+    // The size is based on the height of the crafting GUI and items are always square.
+    private static final float RESULT_ITEM_MAX_SIZE = 0.1f;
+
+    private static final float BUTTON_OK_X = 0.812f;
+    private static final float BUTTON_OK_Y = 0.05f;
+    private static final float BUTTON_OK_WIDTH = 0.15f;
+    private static final float BUTTON_OK_HEIGHT = 0.15f;
+
+    private static final float BUTTON_CANCEL_X = 0.036f;
+    private static final float BUTTON_CANCEL_Y = 0.05f;
+    private static final float BUTTON_CANCEL_WIDTH = 0.15f;
+    private static final float BUTTON_CANCEL_HEIGHT = 0.15f;
+
+    // Colors
+    private static final int NUMBER_BACKGROUND_COLOR = 0xd93030ff;
+
+    private static final Texture texture;
+    private static final TextureRegion numberBackground;
+    private static final Animation backgroundAnimation;
+    private static final BitmapFont bitmapFont;
+
+    static {
+        backgroundAnimation = Animation.of(new File("./game/assets/hud/crafting/idle"));
```

Comment:
`new File` könnte uns in der jar kaputt gehen....
Das müssen wir nachher mal testen

Reply:
In https://github.com/Programmiermethoden/Dungeon/pull/953 hab ich mir tatsächlich die Animation dann selber zusammengeabstelt (also mit den konstruktor von `Animation`)

Reply:
Kein `new File` mehr. "Nur" `Animation.of()`.

Reply:
Rest mach ich als Teil von #953